app/code/community/Clean/SqlReports/Model/Report.php

```diff
@@ -64,4 +64,35 @@ public function getGridConfig()
         }
         return $this->_gridConfig;
     }
-}
\ No newline at end of file
+
+    /**
+     * Disallow TRUNCATE, DROP, DELETE statements & remove semicolon terminator
+     */
+    protected function _beforeSave() {
+        $disallowedPatterns = array(
+            'TRUNCATE TABLE',
+            'DROP TABLE',
+            'DROP TEMPORARY TABLE',
+            'DELETE FROM'
+        );
+		
+        $sqlQuery = $this->getSqlQuery();
+
+        if (substr($sqlQuery, -1) === ';') {
+            $this->setSqlQuery(substr($sqlQuery, 0, -1));
+            Mage::getSingleton('adminhtml/session')->addNotice(Mage::helper('cleansql')->__('Do not include a semicolon terminator'));
+        }
+
+        foreach ($disallowedPatterns as $pattern) {
+            if (stripos($sqlQuery, $pattern) !== false) {
+				$this->_dataSaveAllowed = false;
+				Mage::getSingleton('adminhtml/session')->addError(Mage::helper('cleansql')->__('TRUNCATE, DROP or DELETE statemanets are not allowed'));
+				return $this;
+		    }
+        }
+
+        Mage::getSingleton('adminhtml/session')->addSuccess(Mage::helper('cleansql')->__('Saved report: %s', $this->getTitle()));
```

Comment:
I think this message needs to stay in the controller, it should follow Magento's pattern and errors that display in the frontend really shouldn't come form models if possible. Consider the (not-so-common) scenario were a developer would use the Report model directly outside of Magento's adminhtml context and there is no session.